common/type.h

```diff
@@ -21,6 +21,7 @@
 using tCoreId = uint32_t;
 using tSocketId = uint32_t;
 using tPortId = uint16_t;
+constexpr tPortId INVALID_PORT_ID = std::numeric_limits<tPortId>::max();
```

Comment:
How are we using `INVALID_PORT_ID`? We do not check if the port is valid in any methods. The only usage is marking `m_kni_port` as "Invalid" in the Remove method and then checking its validity in the destructor. But we can achieve the same with a boolean flag and avoid using the maximum u16 value as a marker.

Or is it because some methods __could__ be called on an object with an invalid port, and we want DPDK functions to 100% result in an error? However, if all the code is correct, we shouldn't call such methods on an already invalid object, right?